pkg/controllers/netpol/network_policy_controller.go

```diff
@@ -560,27 +605,31 @@ func (npc *NetworkPolicyController) cleanupStaleIPSets(activePolicyIPSets map[st
 		}()
 	}
 
-	ipsets, err := utils.NewIPSet(false)
-	if err != nil {
-		return fmt.Errorf("failed to create ipsets command executor due to %s", err.Error())
-	}
-	err = ipsets.Save()
-	if err != nil {
-		klog.Fatalf("failed to initialize ipsets command executor due to %s", err.Error())
-	}
-	for _, set := range ipsets.Sets {
-		if strings.HasPrefix(set.Name, kubeSourceIPSetPrefix) ||
-			strings.HasPrefix(set.Name, kubeDestinationIPSetPrefix) {
-			if _, ok := activePolicyIPSets[set.Name]; !ok {
-				cleanupPolicyIPSets = append(cleanupPolicyIPSets, set)
-			}
+	for ipFamily := range npc.ipFamilyHandlers {
```

Comment:
You should do `for _, ipFamily := range npc.ipFamilyHandlers {`, otherwise you pick up the key of the map and not the value, which is what we are doing in the rest of the loops. This helps understand things better

Reply:
Hmmm... nope?

By doing

```
for x := range someMap {
```

you are getting a key in `x`

By doing

```
for _, x := range someMap {
```

you are getting a value in `x`

`npc.ipFamilyHandlers` is a map where `ipFamily uint16` is a key and then struct `ipFamilyHandler` is a value. In that case, I'm interested only in family, not in a handler.

I'm not sure how did you get that compiling on your branch.

Reply:
Ah, it works for you, because you are in fact accessing `ipFamilyHandler.Family`. But I would rather use the key directly. :)

Reply:
Or wait, `ipFamilyHandler.Family` is of different type, coming from k8s, as a string. If we need a string, not uint32, then yes, using `ipFamilyHandler.Family` is the way to go.